spec/models/swift_storage_provider_spec.rb

```diff
@@ -1,13 +1,462 @@
+          .and_return(expected_project_id)
+      end
+
+      context 'project container exists' do
+        let(:expected_meta) {
+          {
+            foo: 'bar',
+            baz: 'stuff'
+          }
+        }
+        it 'should return true' do
+          is_expected.to receive(:get_container_meta)
+            .with(expected_project_id)
+            .and_return(expected_meta)
+          expect {
+            expect(subject.is_initialized?(project)).to be_truthy
+          }.not_to raise_error
+        end
+      end
+
+      context 'project container does not exist' do
+        it 'should return false' do
+          is_expected.to receive(:get_container_meta)
+            .with(expected_project_id)
+          expect {
+            expect(subject.is_initialized?(project)).to be_falsey
+          }.not_to raise_error
+        end
+      end
+
+      context 'unexpected StorageProviderException' do
+        let(:unexpected_exception) { StorageProviderException.new('Unexpected') }
+        it 'should raise the original exception' do
+          is_expected.to receive(:get_container_meta)
+            .with(expected_project_id)
+            .and_raise(unexpected_exception)
+          expect {
+            subject.is_initialized?(project)
+          }.to raise_error(unexpected_exception)
+        end
+      end
+    end
+
+    describe '#single_file_upload_url(upload)' do
+      let(:expected_url) { Faker::Internet.url }
+      it 'should return a signed url to POST the upload' do
+        is_expected.to receive(:build_signed_url)
+          .with(
+            'POST',
+            upload.sub_path,
+            subject.expiry
+          ).and_return(expected_url)
+        expect {
+          expect(subject.single_file_upload_url(upload)).to eq(expected_url)
+        }.not_to raise_error
+      end
+    end
+
+    describe '#initialize_chunked_upload' do
+      it 'should not do anything to initialize a chunked upload in swift' do
+        expect {
+          subject.initialize_chunked_upload(upload)
+        }.not_to raise_error
+      end
+    end
+
+    describe '#endpoint' do
+      it 'should return the swift url_root' do
+        expect {
+          expect(subject.endpoint).to eq(subject.url_root)
+        }.not_to raise_error
+      end
+    end
+
+    describe '#chunk_max_exceeded?' do
+      context 'chunk.upload.chunks.count < chunk_max_number' do
+        it 'should return false' do
+          expect(chunk.upload.chunks.count).to be < subject.chunk_max_number
+          expect(subject.chunk_max_exceeded?(chunk)).to be_falsey
+        end
+      end
+
+      context 'chunk.upload.chunks.count = chunk_max_number' do
+        subject { FactoryBot.create(:swift_storage_provider, chunk_max_number: chunk.upload.chunks.count) }
+        it 'should return true' do
+          expect(chunk.upload.chunks.count).to eq(subject.chunk_max_number)
+          expect(subject.chunk_max_exceeded?(chunk)).to be_truthy
```

Comment:
The context of when this call is being made is when a new chunk is being attempted at the moment when number of stored chunks on the upload is equal to the max. This new chunk should be invalid as it would exceed the chunk_max_number.

Reply:
how about `chunk_max_reached?`

Reply:
Works for me!